### PR TITLE
Topics select field for the UserListFormDialog

### LINKED ISSUE #2395: Add and remove subjects (topics) for a list
As a user, I'd like to add/remove subjects for a list.

#### Designs and Mockups

The list form should include a multiselect dropdown of available topics, and buttons with x's for selected topics (clicking the x will remove the topic).  Users should not be able to create their own topics.

@aliceriot do you know of an npm module that provides a component like the above?

#### Acceptance Criteria:

- [x] API supports viewing, adding, removing list categories
- [ ] UI for viewing, adding, removing list categories

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #2395 

#### What's this PR do?
- Adds a multi-select field for topics on the UserListFormDialog

#### How should this be manually tested?
- Create new userlists with and without topics
- Edit userlists - change, add, remove topics

#### Screenshots (if appropriate)
Screen Shot 2019-11-21 at 12 35 06 PM

### CODE REVIEW

**Comment by aliceriot** [comment]
two styling things I noticed:

fffejejejeje

- in the designs the name for the field (the text shown above it) is just 'Subjects', not 'List Subjects'
- I think we might want to put 10-15px of padding below the subjects input, so there's some visual separation between that an the `border-bottom` on the `.user-list-create-form` div

**Comment by aliceriot** [comment]
but as far as the functionality everything works great! I can add and remove subjects no problem.

I do have a question about the usage of react-select though, it seems on my machine anyhow there is some lag that happens when you type into the field which is really sort of strange. Here's a gif where I tried to capture it:

topbfdfdfdf

idk if that fully captures it, but it's sort of weird and disconcerting. On the [component's website](https://react-select.com/) the examples don't do this, so I'm wondering what we're doing that's leading to this behavior - maybe something about the interplay between Formik and react-select? in any case I think we should figure it out and eliminate it.

**Comment by Ferdi** [comment]
@mattbert is it hard to limit nr of item user can select?

**Comment by mbertrand** [comment]
@Ferdi it should be straightforward.  What should the limit be?

**Comment by aliceriot** [comment]
@mbertrand is this ready for review again?

**Comment by mbertrand** [comment]
Yes

File: static/js/lib/queries/topics.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// @flow
+
+import { topicApiURL } from "../url"
+import { merge } from "ramda"
+import type { CourseTopic } from "../../flow/discussionTypes"
+import { constructIdMap } from "../redux_query"
+import { createSelector } from "reselect"
```
Comment (aliceriot) [inline static/js/lib/queries/topics.js]: code formatting nit: can you remove the blank line below `// @flow` and separate the `type` import from the rest with a blank line?

File: static/js/lib/queries/topics_test.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// @flow
+
```
Comment (aliceriot) [inline static/js/lib/queries/topics_test.js]: another nit: blank line :)

File: static/js/components/Select_test.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  beforeEach(() => {
+    sandbox = sinon.createSandbox()
+    closeMenuOnSelect = false
+    openMenuOnClick = false
+    isMulti = false
+    placeholder = "Select something..."
+    menuPlacement = "bottom"
+    className = "class"
+    field = { name: "TestName" }
+    ;(form = {}),
```
Comment (aliceriot) [inline static/js/components/Select_test.js]: there's a weird formatting thing here (prettier artifact?)

can change to just `form = {}` and `options = [ ... ]`

File: static/js/lib/queries/topics.js (hunk 1 above)
Comment (mbertrand) [reply static/js/lib/queries/topics.js]: ✔️

**Comment by mbertrand** [comment]
@aliceriot ready for review again

**Pull request merged**